src/dbspro/cli/run.py

```diff
@@ -26,7 +26,7 @@ def main(args):
         logging.info(f'Output directory {args.directory} created.')
 
     # Append full path to targets.
-    targets_with_path = [f"{args.directory}/{t}" for t in args.targets]
+    targets_with_path = [f"{args.directory}/{t}" for t in args.targets] if args.targets else None
```

Comment:
`else None` can be omitted (or possibly instead write `if args.targets: targets_with_path ...`), not 100% it works as a one-liner.

Reply:
Yeah but since I am defining a new variable I still need to give it something. See following test:

```python
>>> targets = ["test.txt", "test2.txt"]
>>> targets_with_path = [f"home/{t}" for t in targets] if targets
  File "<stdin>", line 1
    targets_with_path = [f"home/{t}" for t in targets] if targets
                                                                ^
SyntaxError: invalid syntax
>>> targets_with_path = [f"home/{t}" for t in targets] if targets else None
>>> targets_with_path
['home/test.txt', 'home/test2.txt']
````

I could however do this: 

```python
targets_with_path=None
if args.targets:
    targets_with_path = [...list..]
```

But then I can just as well do my one-liner. The only reason not to might be that this is easier to read.

Reply:
Fair enough.